src/v1/Models/Item.php

```diff
@@ -46,6 +48,23 @@ class Item extends Model
     // 'getItems'
   ];
 
+  public static function boot()
+  {
+    parent::boot();
+    static::creating(function ($model)
+    {
+      /**
+       * Manage id_bytype field
+       * Get max id in DB for this type and increase 1 to get new id_bytype
+       */
+      $subquery = DB::table((new self())->getTable() . ' as item_alias')
+        ->selectRaw('coalesce(max(id_bytype), 0) + 1 as id_bytype')
+        ->whereRaw('type_id=' . $model->type_id)
```

Comment:
Let's do our best to avoid SQL injections:

```suggestion
        ->where('type_id', '=',  $model->type_id)
```

Reply:
I fixed, can you do the review again because I modify the next line too to works :D